### PR TITLE
PSR-269 PSR QUERY 2 : handling recordVersion flag

### PR DESCRIPTION
**Pull request by cyeniceri**

### CODE REVIEW

File: test/resources/PSR with Login/Standard PSR - Bonds Only - Success.bru [hunk 1]
```diff
@@ -16,72 +16,75 @@ auth:bearer {
...
-                      "methodOfHolding": "02",
-                      "optDateOfAcqOrContrib": "2022-07-30",
-                      "costOfBonds": 2000.5,
-                      "optConnectedPartyStatus": false,
-                      "bondsUnregulated": false,
-                      "totalIncomeOrReceipts": 300
-                  }
-              ]
+      "schemeDesignatory": {
+        "openBankAccount": true,
```
Comment (kberec) [inline test/resources/PSR with Login/Standard PSR - Bonds Only - Success.bru:35]: As a matter of interest, why only for accountingPeriodDetails? Should it be a record version value with some other sections too where it exists in the schema?

File: app/uk/gov/hmrc/pensionschemereturn/connectors/PsrConnector.scala [hunk 2]
```diff
@@ -178,7 +178,7 @@ class PsrConnector @Inject()(config: AppConfig, http: HttpClient)
         case SERVICE_UNAVAILABLE =>
```
Comment (kberec) [inline app/uk/gov/hmrc/pensionschemereturn/connectors/PsrConnector.scala:178]: Can this PR get merged without the second part (FE changes)?

File: test/resources/PSR with Login/Standard PSR - Bonds Only - Success.bru (hunk 1 above)
Comment (cyeniceri) [reply test/resources/PSR with Login/Standard PSR - Bonds Only - Success.bru:35]: I added recordVersion for rest of them in Full Body - Success collection. In other collections, recordVersion(s) are None to test when omitted.

**Review by kberec — APPROVED** [review]
We had a detailed call with the team to go through these changes, and all are happy to get this merged.

**Pull request merged**